Sources/ApolloCodegenLib/Templates/MockObjectTemplate.swift

```diff
@@ -76,10 +76,18 @@ struct MockObjectTemplate: TemplateRenderer {
   
   private func mockFunctionDescriptor(_ graphQLType: GraphQLType) -> String {
     switch graphQLType {
-      case .list(.nonNull(.scalar(_))):
-        return "ScalarList"
-      case .list(_):
-        return "List"
+      case .list(let type):
+        switch type {
+        case .nonNull(.list(_)),
+             .list(_):
+          return mockFunctionDescriptor(type)
+        case .nonNull(.entity(_)),
+             .entity(_):
+          return "List"
+        default:
+          break
```

Comment:
Should the `return "ScalarList"` just be here in the `default` case instead of the `break`?

Reply:
It could go there there as well, but it would still need to be where it is now so it would be in two places because of the nested switch statement so I just had it break out of the inner switch and have a single return for `ScalarList`

Reply:
ooooh i see! Nevermind this is fine!